src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -258,15 +258,35 @@ public Screen getNextScreen(boolean needsFullEntityScreen, EntityScreenContext e
             queryScreen.init(sessionWrapper);
             return queryScreen;
         } else if (next.equalsIgnoreCase(SessionFrame.STATE_SYNC_REQUEST)) {
-            String username = session.getAsUser() != null ?
-                    StringUtils.getFullUsername(session.getAsUser(), session.getDomain()) : null;
-            FormplayerSyncScreen syncScreen = new FormplayerSyncScreen(username);
-            syncScreen.init(sessionWrapper);
-            return syncScreen;
+            return getSyncScreen();
         }
         throw new RuntimeException("Unexpected Frame Request: " + sessionWrapper.getNeededData());
     }
 
+    public Screen getNextScreenIfSyncScreen(boolean needsFullEntityScreen, EntityScreenContext entityScreenContext) throws CommCareSessionException {
+        String next = sessionWrapper.getNeededData(sessionWrapper.getEvaluationContext());
+        if (next == null) {
+            if (sessionWrapper.isViewCommand(sessionWrapper.getCommand())) {
+                sessionWrapper.stepBack();
```

Comment:
Can this ever cause an infinite loop? "step back", next is null, "step back"

Reply:
I don't think so, this logic is the same as what's done for `getNextScreen`. And from what i'm understanding, `stepBack` pops steps until we reach a step that requires user input, at which point `isViewCommand` will be false. 

But actually, I think this can be removed since `syncScreen` does not requires user input so this will not result in a syncScreen.

Reply:
[f6c4d59](https://github.com/dimagi/formplayer/pull/1608/commits/f6c4d59baa7ceb5f19a6088c28e6abb1f3c8239e)